src/distanceMatrix.js

```diff
@@ -0,0 +1,121 @@
+import { SymmetricMatrix } from './symmetricMatrix';
+
+export class DistanceMatrix extends SymmetricMatrix {
+  /**
+   * not the same as matrix.isSymmetric()
+   * Here is to check if it's instanceof SymmetricMatrix without bundling issues
+   *
+   * @param value
+   * @returns {boolean}
+   */
+  static isDistanceMatrix(value) {
+    return (
+      SymmetricMatrix.isSymmetricMatrix(value) &&
+      value.klassSubType === 'DistanceMatrix'
+    );
+  }
+
+  constructor(sideSize) {
+    super(sideSize);
+
+    if (!this.isDistance()) {
+      throw new TypeError('Provided arguments do no produce a distance matrix');
+    }
+  }
+
+  set(rowIndex, columnIndex, value) {
+    // distance matrix diagonal is 0
+    if (rowIndex === columnIndex) value = 0;
```

Comment:
I had in mind, the static helpers like `.fill()` and because AbstractMatrix have (I suppose) a lot of this kind of in-place-mutation. I go to this solution.